lib/puppet/resource/capability_finder.rb

```diff
@@ -0,0 +1,61 @@
+#
+# A helper module to look a capability up from PuppetDB
+#
+# @todo lutter 2015-03-10: determine whether this should be based on
+# Puppet::Pops::Evaluator::Collectors, or at least use
+# Puppet::Util::Puppetdb::Http
+
+require 'net/http'
+require 'cgi'
+
+module Puppet::Resource::CapabilityFinder
+
+  # Look the capability resource with the given +type+ and +title+ up from
+  # PuppetDB.
+  def self.find(environment, cap)
+    http = Net::HTTP.new("localhost", 8080)
+    query = ["and", ["=", "type", cap.type.capitalize],
+                    ["=", "title", cap.title.to_s],
+                    ["=", "tag", "producer:#{environment}"]].to_json
+
+    Puppet.notice "Capability lookup #{cap}]: #{query}"
+    response = http.get("/pdb/query/v4/resources?query=#{CGI.escape(query)}",
+                        { "Accept" => 'application/json'})
+
+    json = response.body
+
+    # @todo lutter 2014-11-04: use just JSON
+    data = PSON.parse(json)
```

Comment:
this parsing is unprotected and will produce a bad error message if there is something wrong with the response body in json. Need to catch errors from that and fail the same way (probably) as if an array was not produced.